tsdb/engine/tsm1/engine.go

```diff
@@ -1117,6 +1117,10 @@ func (e *Engine) WriteSnapshot() error {
 		return err
 	}
 
+	if snapshot.Size() == 0 {
+		return nil
```

Comment:
I think you will also need to run `e.Cache.ClearSnapshot(true)` in this case.  That is currently done in `e.writeSnapshotAndCommit`.